### DISCUSSION TITLE
anyway to have the factory samples and factory patterns match up after re installing them

### DISCUSSION THREAD

**Discussion by Bornin92** [bangcorrupt/hacktribe#61] [General]
is there anyway to have the factory samples and factory patterns match up after re installing them from e2samples.all file?

**Comment by bangcorrupt** [comment]
It should only require adding 481 to the sample index for every part in every pattern file.  Only if the existing sample index is >19 (i.e. a sample not an oscillator).  I can write a script to test this.

**Reply by Bornin92** [reply]
that would be great ! , I've got all my factory samples starting at 501. but could change to 481 to match what you're making ! whatever works

**Reply by ecco00** [reply]
pls share the factory e2samples.all :)

**Reply by bangcorrupt** [reply]
See [Factory Samples](../wiki/How-To#factory-samples) in the wiki.

**Reply by Smarnz** [reply]
This has been on my mind and so glad it's been asked and answered. Did you write a script that does this? Ideally for every part in every pattern from 001-150 so it does not mess with the user patterns but whatever is possible :) thanks so much this hack makes this the ideal device

**Reply by bangcorrupt** [reply]
Not yet, but keep an eye on [e2-scripts #8](https://github.com/bangcorrupt/e2-scripts/issues/8).

**Comment by Smarnz** [comment]
Will keep on the pulse. Much appreciated

**Comment by bangcorrupt** [comment]
@untergeekDE  added a [script](https://github.com/bangcorrupt/hacktribe/blob/main/scripts/e2pat_shift.py) to shift sample references in an allpattern file.

`python e2pat_shift.py -i input.e2sallpat -o output.e2sallpat -s 482`
